runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java

```diff
@@ -25,109 +25,181 @@
+	SLIMY_EEL("Bait (Swamp)", "Slimy Eel", new LinkedHashMap<>(){{
+		put(FishingTools.FISHING_ROD, new Catches("Slimy Eel", ItemID.RAW_SLIMY_EEL));
+	}},
 		FISHING_SPOT_2653, FISHING_SPOT_2654, FISHING_SPOT_2655
 	),
-	DARK_CRAB("Dark Crab", ItemID.RAW_DARK_CRAB,
+	DARK_CRAB("Cage/Harpoon (Wilderness)", "Dark Crab", new LinkedHashMap<>(){{
+		put(FishingTools.LOBSTER_POT, new Catches("Dark Crab, Lobster", ItemID.RAW_DARK_CRAB));
+		put(FishingTools.HARPOON, new Catches("Tuna, Swordfish", ItemID.RAW_SWORDFISH));
+	}},
 		FISHING_SPOT_1535, FISHING_SPOT_1536
 	),
-	COMMON_TENCH("Common tench, Bluegill, Greater siren, Mottled eel", "Greater siren", ItemID.COMMON_TENCH,
+	COMMON_TENCH("Catch (Aerial Fishing)", "Aerial Fishing", new LinkedHashMap<>(){{
+		put(FishingTools.CORMORANTS_GLOVE, new Catches("Bluegill, Common Tench, Mottled Eel, Greater Siren", ItemID.COMMON_TENCH));
+	}},
 		FISHING_SPOT_8523
 	),
-	TUTORIAL_SHRIMP("Shrimp", ItemID.RAW_SHRIMPS,
+	TUTORIAL_SHRIMP("Fish (Tutorial Island)", "Shrimp", new LinkedHashMap<>(){{
+		put(FishingTools.SMALL_FISHING_NET, new Catches("Your first catch", ItemID.RAW_SHRIMPS));
+	}},
 		FISHING_SPOT_3317
 	),
-	ETCETERIA_LOBSTER("Lobster", "Lobster (Approval only)", ItemID.RAW_LOBSTER,
+	ETCETERIA_LOBSTER("Cage/Harpoon (Approval only)", "Lobster (Approval only)", new LinkedHashMap<>(){{
+		put(FishingTools.LOBSTER_POT, new Catches("Lobster (Approval only)", ItemID.RAW_LOBSTER));
+		put(FishingTools.HARPOON, new Catches("Tuna, Swordfish (Approval only)", ItemID.RAW_SWORDFISH));
+	}},
 		FISHING_SPOT_3657
 	),
-	QUEST_RUM_DEAL("Sluglings", "Rum deal (Quest)", ItemID.SLUGLINGS,
+	QUEST_RUM_DEAL("Fish (Braindeath Island)", "Rum Deal (Quest)", new LinkedHashMap<>(){{
+		put(FishingTools.FISHBOWL_NET, new Catches("Sluglings, Karamthulhu", ItemID.SLUGLINGS));
+	}},
 		FISHING_SPOT
 	),
-	QUEST_TAI_BWO_WANNAI_TRIO("Karambwan", "Tai Bwo Wannai Trio (Quest)", ItemID.RAW_KARAMBWAN,
+	QUEST_TAI_BWO_WANNAI_TRIO("Fish (Tai Bwo Wannai Trio)", "Tai Bwo Wannai Trio (Quest)", new LinkedHashMap<>(){{
+		put(FishingTools.KARAMBWAN_VESSEL, new Catches("Karambwan", ItemID.RAW_KARAMBWAN));
+	}},
 		FISHING_SPOT_4714
 	),
-	QUEST_FISHING_CONTEST("Giant carp", "Fishing Contest (Quest)", ItemID.GIANT_CARP,
+	QUEST_FISHING_CONTEST("Fish (Hemenster)", "Fishing Contest (Quest)", new LinkedHashMap<>(){{
+		put(FishingTools.FISHING_ROD, new Catches("Giant Carp", ItemID.RAW_GIANT_CARP));
+	}},
 		FISHING_SPOT_4079, FISHING_SPOT_4080, FISHING_SPOT_4081, FISHING_SPOT_4082
 	),
 	;
 
+	private static class Catches{
+		private final String fishName;
+		private final int fishSpriteId;
+
+		Catches(String fish, int spriteId) {
+			this.fishName = fish;
+			this.fishSpriteId = spriteId;
+		}
+	}
+
 	private static final Map<Integer, FishingSpot> SPOTS;
 
 	private final String name;
 	private final String worldMapTooltip;
-	private final int fishSpriteId;
+	private final LinkedHashMap<FishingTools, Catches> catchMap;
```

Comment:
This can just be a `Map` and use `ImmutableMap`s intead, as those also have deterministic iteration. That said, I'm not certain a map is the right solution for this feature--I will think on it and follow up on this comment later.